app/src/views/GrantRoundDetails.vue

```diff
@@ -0,0 +1,254 @@
+            errorMsg="Please enter a valid address"
+          />
+
+          <!-- contribution amount -->
+          <BaseInput
+            v-model="form.amount"
+            description="The number of tokens to contribute"
+            id="contribution-amount"
+            label="Contribution amount"
+            :rules="(amount) => amount || 0 > 0"
+            errorMsg="Please enter an amount greater than 0"
+          />
+
+          <!-- Submit and cancel buttons -->
+          <button
+            type="submit"
+            class="btn btn-primary w-full"
+            :class="{ 'btn-primary-disabled': !isFormValid }"
+            :disabled="!isFormValid"
+          >
+            Send Contribution
+          </button>
+          <button @click.prevent="cancelContribution" class="btn btn-outline w-full">Cancel</button>
+        </form>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref } from 'vue';
+import { useRoute } from 'vue-router';
+import BaseInput from 'src/components/BaseInput.vue';
+// --- Store ---
+import useDataStore from 'src/store/data';
+import useWalletStore from 'src/store/wallet';
+// --- Methods and Data ---
+import { GRANT_ROUND_ABI, ERC20_ABI } from 'src/utils/constants';
+import { BigNumber, Contract, ContractTransaction } from 'src/utils/ethers';
+import { daysAgo, formatAddress, isValidAddress, isValidUrl } from 'src/utils/utils';
+// --- Types ---
+import { GrantRound } from '@dgrants/types';
+import { GrantRound as GrantRoundContract } from '@dgrants/contracts';
+import { BigNumberish, utils } from 'ethers';
+
+function useGrantRoundDetail() {
+  const { grantRounds, poll } = useDataStore();
+  const { signer, userAddress } = useWalletStore();
+  const route = useRoute();
+
+  // get a single grantRound or an empty/error object (TODO: should the typings be modified to account for an empty object?)
+  const grantRound = computed(() => {
+    if (grantRounds.value) {
+      // filter for a matching GrantRound
+      const round = grantRounds.value.filter((round) => round.address == route.params.address);
+
+      return <GrantRound>(round.length ? round[0] : { error: `No GrantRound @ ${route.params.address}` });
+    } else {
+      return <GrantRound>{};
+    }
+  });
+
+  // --- Contribution capabilities ---
+  const isContributing = ref(false);
+  const form = computed<{ token: string; amount: BigNumberish }>(() => {
+    return {
+      token: grantRound.value.donationToken,
+      amount: grantRound.value.minContribution,
+    };
+  });
+  const isFormValid = computed(() => {
+    const { token, amount } = form.value;
+    const areFieldsValid =
+      isValidAddress(<string>token) && token == grantRound.value.donationToken && (amount || 0 > 0);
+    return areFieldsValid;
+  });
+
+  /**
+   * @notice Check against the grantRounds status for a match
+   */
+  const hasStatus = (status: string) => (round: GrantRound) => round.status == status;
```

Comment:
Just curious, why this syntax instead of `const hasStatus = (status: string, round: GrantRound) => round.status == status` ?

Reply:
Ahh, I did that to tidy up the filtering over in `GrantRoundsList.vue` and copied the same fn into this file. Would it make sense to keep this in `utils.ts` - I'm not sure its generic enough?

We could just do as you say here, theres no need for currying with the way we're using it.

Reply:
Since it's used in two different places, I think it's generic enough to move into the `utils.ts` file!

Reply:
Resolved in 85d0d88